master/buildbot/test/unit/test_steps_trigger.py

```diff
@@ -542,3 +550,23 @@ def test_waitForFinish_interrupt(self):
         self.step.interrupt(failure.Failure(RuntimeError('oh noes')))
 
         return d
+
+    def test_getSchedulersAndProperties_back_comp(self):
+        class DynamicTrigger(trigger.Trigger):
+            def getSchedulersAndProperties(self):
+                return [("a", {}, False), ("b", {}, True)]
+
+        self.setupStep(DynamicTrigger(schedulerNames=['a', 'b']))
+        self.scheduler_a.result = SUCCESS
+        self.scheduler_b.result = FAILURE
+        self.expectOutcome(result=SUCCESS, state_string='triggered a, b')
+        self.expectTriggeredWith(a=(False, [], {}), b=(False, [], {}))
+        return self.runStep()
+
+    def test_unimportantsShedulerNames(self):
+        self.setupStep(trigger.Trigger(schedulerNames=['a', 'b'], unimportantSchedulerNames=['b']))
+        self.scheduler_a.result = SUCCESS
+        self.scheduler_b.result = FAILURE
+        self.expectOutcome(result=SUCCESS, state_string='triggered a, b')
+        self.expectTriggeredWith(a=(False, [], {}), b=(False, [], {}))
+        return self.runStep()
```

Comment:
You can just create another scheduler 'c' in SetupStep which only one unit test will use